src/plugins/data/public/search/search_service.test.ts

```diff
@@ -155,90 +155,6 @@ describe('Search service', () => {
-              type: 'other_kind_of_exception',
-              reason: 'reason of other_kind_of_exception',
-            },
-          },
-          { reason: { type: 'fatal_warning', reason: 'this is a fatal warning message' } },
-        ] as unknown as estypes.ShardFailure[];
-
-        const responder = inspector.adapter.start('request1');
-        responder.ok(getMockResponseWithShards(shards));
-        data.showWarnings(inspector.adapter, callback);
-
-        expect(notifications.toasts.addWarning).toBeCalledTimes(1);
-        expect(notifications.toasts.addWarning).toBeCalledWith({
-          title: '2 of 4 shards failed',
-          text: expect.any(Function),
-        });
-      });
-
-      it('can show a timed_out warning', () => {
-        const responder = inspector.adapter.start('request1');
-        shards = { total: 4, successful: 4, skipped: 0, failed: 0 };
-        const response1 = getMockResponseWithShards(shards);
-        response1.json.rawResponse.timed_out = true;
-        responder.ok(response1);
-        data.showWarnings(inspector.adapter, callback);
-
-        expect(notifications.toasts.addWarning).toBeCalledTimes(1);
-        expect(notifications.toasts.addWarning).toBeCalledWith({
-          title: 'Data might be incomplete because your request timed out',
-        });
-      });
-
-      it('can show two warnings if response has shard failures and also timed_out', () => {
-        const responder = inspector.adapter.start('request1');
-        const response1 = getMockResponseWithShards(shards);
-        response1.json.rawResponse.timed_out = true;
-        responder.ok(response1);
-        data.showWarnings(inspector.adapter, callback);
-
-        expect(notifications.toasts.addWarning).toBeCalledTimes(2);
-        expect(notifications.toasts.addWarning).nthCalledWith(1, {
-          title: 'Data might be incomplete because your request timed out',
-        });
-        expect(notifications.toasts.addWarning).nthCalledWith(2, {
-          title: '2 of 4 shards failed',
-          text: expect.any(Function),
-        });
-      });
-
-      it('will show multiple warnings when multiple responses have shard failures', () => {
-        const responder1 = inspector.adapter.start('request1');
-        const responder2 = inspector.adapter.start('request2');
-        responder1.ok(getMockResponseWithShards(shards));
-        responder2.ok({
-          json: {
-            rawResponse: {
-              timed_out: true,
-            },
-          },
-        });
-
-        data.showWarnings(inspector.adapter, callback);
-
-        expect(notifications.toasts.addWarning).toBeCalledTimes(2);
-        expect(notifications.toasts.addWarning).nthCalledWith(1, {
-          title: '2 of 4 shards failed',
-          text: expect.any(Function),
-        });
-        expect(notifications.toasts.addWarning).nthCalledWith(2, {
-          title: 'Data might be incomplete because your request timed out',
-        });
-      });
```

Comment:
Which of these test cases should be kept?

"will show single notification when some warnings are filtered". Removed test case because there is a single warning for incomplete data instead of separate warnings per shard failure

"can show a timed_out warning". Removed this test case because time_out warning removed

"can show two warnings if response has shard failures and also timed_out" Removed test case because there is a single warning for incomplete data instead of treating these as separate warnings

"will show multiple warnings when multiple responses have shard failures" This is an interesting one, and the description does not match the implementation. The implementation sends a shard warning and a timeout warning.

Reply:
Okay, thanks for explaining!